rs/boundary_node/anonymization/client/src/lib.rs

```diff
@@ -251,3 +259,174 @@ impl Submit for Canister {
         }
     }
 }
+
+// Canister methods
+
+pub struct CanisterMethods {
+    /// register method tied to canister
+    register: Arc<dyn Register>,
+
+    /// query method tied to canister
+    query: Arc<dyn Query>,
+
+    /// submit method tied to canister
+    submit: Arc<dyn Submit>,
+}
+
+impl From<Canister> for CanisterMethods {
+    fn from(value: Canister) -> Self {
+        Self {
+            register: Arc::new(value.clone()),
+            query: Arc::new(value.clone()),
+            submit: Arc::new(value.clone()),
+        }
+    }
+}
+
+// Client
+
+#[async_trait]
+pub trait Track: Sync + Send {
+    async fn track(&mut self, cb: impl Fn(Vec<u8>) + Send + Sync) -> Result<(), Error>;
```

Comment:
Would it make sense to make this trait immutable and simply return the salt, and have another struct e.g. `PeriodicTracker`, which would execute this `track()` in a loop?

Reply:
Can you clarify what the ask here? What issue is this solving?

Reply:
There are no real issues, it's rather a friendly suggestion, IMO it would make the following improvements:
- Returning something from the function instead of mutating implementor's state is generally cleaner/safer and reduces unexpected side effects
- Moving the loop outside of the function will make it a bit more readable, as now the `loop` covers a rather big code chunk of code and one needs to find/understand which exact parts are retried
- Will also improve testability, as You could just test one function execution, instead of smth infinite